src/node/index.js

```diff
@@ -55,73 +55,98 @@ var grpc = require('./src/grpc_extension');
 grpc.setDefaultRootsPem(fs.readFileSync(SSL_ROOTS_PATH, 'ascii'));
 
 /**
- * Load a gRPC object from an existing ProtoBuf.Reflect object.
- * @param {ProtoBuf.Reflect.Namespace} value The ProtoBuf object to load.
+ * Build a gRPC object from an existing ProtoBuf.Root object.
+ * @param {ProtoBuf.Root} value The ProtoBuf root object to transform.
  * @param {Object=} options Options to apply to the loaded object
- * @return {Object<string, *>} The resulting gRPC object
+ * @return {ProtoBuf.Root} The transformed Root object.
  */
 exports.loadObject = function loadObject(value, options) {
-  var result = {};
-  if (value.className === 'Namespace') {
-    _.each(value.children, function(child) {
-      result[child.name] = loadObject(child, options);
-    });
-    return result;
-  } else if (value.className === 'Service') {
-    return client.makeProtobufClientConstructor(value, options);
-  } else if (value.className === 'Message' || value.className === 'Enum') {
-    return value.build();
-  } else {
+  if (!value) {
     return value;
   }
+
+  // Protobuf.JS 6 trees don't work the same way as in ProtoBuf.JS 5.
+  // Now, we can just swap out the Services in the tree.
+  if (value.methods && Object.keys(value.methods).length) {
+    value.resolveAll();
+    return client.makeProtobufClientConstructor(value, options);
+  }
+
+  if (value.nested) {
+    for (let name in value.nested) {
+      if (!value.nested.hasOwnProperty(name)) {
+        continue;
+      }
+      value.nested[name] = loadObject(value.nested[name], options);
+    }
+  }
+
+  return value;
 };
 
 var loadObject = exports.loadObject;
 
 /**
+ * Compat: apply a ProtoBuf.JS 5 {root, file} object
+ * to a ProtoBuf.Root.
+ * @param {Object=} filename Filename object.
+ * @param {ProtoBuf.Root=} ProtoBuf.Root to apply to.
+ * @return {string} The filename to pass to ProtoBuf.load()
+ */
+function applyProtoRoot(filename, root) {
+  if (_.isString(filename)) {
+    return filename;
+  }
+  filename.root = path.resolve(filename.root) + '/';
+  root.resolvePath = function(originPath, importPath, alreadyNormalized) {
+    return ProtoBuf.util.resolvePath(filename.root,
+      importPath,
+      alreadyNormalized);
+  };
+  return filename.file;
+};
+
+/**
  * Load a gRPC object from a .proto file. The options object can provide the
  * following options:
- * - convertFieldsToCamelCase: Loads this file with that option on protobuf.js
- *   set as specified. See
- *   https://github.com/dcodeIO/protobuf.js/wiki/Advanced-options for details
+ * - enumsAsStrings: deserialize enum values as strings instead of numeric
+ *   values. Defaults to false.
+ * - longsAsStrings: deserialize long values as strings instead of objects.
+ *   Defaults to true.
+ * @param {string} filename The file to load
+ * @param {Object=} options Options to apply to the loaded file
+ * @return {ProtoBuf.Root=} ProtoBuf.Root with services filled.
+ */
+exports.load = function load(filename, options) {
```

Comment:
Please do not change the signature of `grpc.load`. Even if `format` is unused, we need to keep it for API compatibility.